app/controllers/users.js

```diff
@@ -1,7 +1,40 @@
-const { createUser } = require('../services/users');
+const { createUser, signInUser } = require('../services/users');
+const jwt = require('jwt-simple');
 const lodash = require('lodash');
+const logger = require('../logger');
+const {
+  common: {
+    session: { secret }
+  }
+} = require('../../config');
+const errors = require('../errors');
+const bcrypt = require('bcryptjs');
 
 exports.createUser = (req, res, next) =>
   createUser(req.body)
     .then(user => res.send(lodash.pick(user, ['name', 'last_name', 'email'])))
     .catch(next);
+
+exports.signIn = (req, res, next) =>
+  signInUser(req.body)
+    .then(user => {
+      if (user) {
+        return bcrypt
+          .compare(req.body.password, user.password)
+          .then(passwordExists => {
+            if (!passwordExists) {
+              next(errors.badRequestError('Wrong password!'));
+            }
+            logger.info('User exists and passwords are matching');
+            const token = jwt.encode({ username: req.body.email }, secret);
```

Comment:
encode is not a promise?

Reply:
Based in the documentation, is not a promise https://www.npmjs.com/package/jwt-simple